OpenRA.Game/MiniYaml.cs

```diff
@@ -254,25 +254,25 @@ static List<MiniYamlNode> FromLines(IEnumerable<ReadOnlyMemory<char>> lines, str
 					if (commentStart >= 0 && !discardCommentsAndWhitespace)
 						comment = line.Slice(commentStart);
 
-					// Remove leading/trailing whitespace guards
 					if (value.Length > 1)
 					{
+						// Remove leading/trailing whitespace guards
 						var trimLeading = value[0] == '\\' && (value[1] == ' ' || value[1] == '\t') ? 1 : 0;
 						var trimTrailing = value[value.Length - 1] == '\\' && (value[value.Length - 2] == ' ' || value[value.Length - 2] == '\t') ? 1 : 0;
 						if (trimLeading + trimTrailing > 0)
 							value = value.Slice(trimLeading, value.Length - trimLeading - trimTrailing);
-					}
 
-					// Remove escape characters from #
-					if (value.Contains("\\#", StringComparison.Ordinal))
-						value = value.ToString().Replace("\\#", "#");
+						// Remove escape characters from #
+						if (value.Contains("\\#", StringComparison.Ordinal))
+							value = value.ToString().Replace("\\#", "#");
+					}
 				}
 
-				if (key != null || !discardCommentsAndWhitespace)
+				if (!key.IsEmpty || !discardCommentsAndWhitespace)
 				{
-					var keyString = key == null ? null : key.ToString();
-					var valueString = value == null ? null : value.ToString();
-					var commentString = comment == null ? null : comment.ToString();
+					var keyString = key.IsEmpty ? null : key.ToString();
+					var valueString = value.IsEmpty ? null : value.ToString();
+					var commentString = comment == default ? null : comment.ToString();
```

Comment:
Can we please add a comment like
```csharp
// Note: We need to support empty comments here to ensure that empty comments
// (i.e. a lone # at the end of a line) can be correctly re-serialized
```

here? The need for this difference wasn't obvious until I spent some time poking at it.